src/components/ProposalDetail/ProposalActions.js

```diff
@@ -77,12 +80,15 @@ function ProposalActions({
         mode: 'strong',
       }
     }
-    return {
-      text: 'Support this proposal',
-      action: onRequestSupportProposal,
-      mode: 'strong',
-      disabled: !accountBalance.gt(0),
+    if (mode === 'support') {
+      return {
+        text: 'Support this proposal',
+        action: onRequestSupportProposal,
+        mode: 'strong',
+        disabled: !accountBalance.gt(0),
+      }
     }
+    return null
```

Comment:
Should we place it at the top to avoid all the if statements?

```suggestion
   if(!mode) {
     return null
   }
```